### ISSUE TITLE
Finish Keras integration

### ISSUE THREAD

**Issue by ClimbsRocks** [ClimbsRocks/auto_ml#169]
1. Tests!

  - [ ] Duplicate our existing api coverage tests for Keras
  - [ ] Install Keras as part of our travis-ci build

2. Fix how we handle splitting the data formatting pipeline and the model training step. Right now I think it only works if we have `optimize_final_model=True`
3. Move the build_fn functions to utils_models
4. Make sure we have hyperparameters to optimize for classifiers and regressors
5. Make sure our build functions work for classifiers and regressors
6. Make a note somewhere that we only support binary classification and regression, not multi-label classification. 
7. Add to documentation
8. Add to https://github.com/ChristosChristofidis/awesome-deep-learning
